app/routes.py

```diff
@@ -95,3 +95,41 @@ def get_id(kml_admin_id):
         ),
         200
     )
+
+
+@app.route('/kml/<kml_admin_id>', methods=['DELETE'])
+def delete_id(kml_admin_id):
+    enforcer = DynamoDBFilesHandler(
+        table_name=AWS_DB_TABLE_NAME,
+        bucket_name=AWS_S3_BUCKET_NAME,
+        table_region=AWS_DB_REGION_NAME,
+        endpoint_url=AWS_DB_ENDPOINT_URL
+    )
+    item = enforcer.get_item(kml_admin_id)
+
+    # Fetching a non existing Item will return "None"
+    if item is None:
+        logger.error("Could not find the following kml id in the database: %s", kml_admin_id)
+        abort(400, f"Could not find {kml_admin_id} within the database.")
+
+    file_id = item['file_id']
+
+    executor = S3FileHandling(AWS_S3_REGION_NAME, AWS_S3_ENDPOINT_URL)
+    executor.delete_file_in_bucket(AWS_S3_BUCKET_NAME, file_id)
+
+    enforcer.delete_item(kml_admin_id)
```

Comment:
Probably it makes sense to change the order of these two lines?
As it is now, in case the deletion of the file in S3 works but the deletion in the DynamoDB table fails for whatever reason, then the fkml_admin_id would still be found in the DB but no corresponding file in the s3 bucket would be found.

The other way round would not be that bad:
After changing the order, when the deletion in the DynamoDB succeeds, but the deletion of the file in s3 fails, then the kml_admin_id will not be found in the DB and hence no access to the still existing file on s3 will ever be made.

Or even better use a `try ... except` for these two lines?
What do you think @ltshb?

Reply:
Here we need to thing of the error case that can happen and what to do in those error cases:

1. Cannot delete file on bucket, because file is not present (Could happen if we manually delete the files on S3) => fine go ahead we anyway wanted to delete the file
2. Cannot delete file on bucket, because of network issues and/or S3 service unreachable or any other exceptions => stop deletion with 500 http code, but leave in dynamo db. For this case don't do anything at this level let the application crash (make sure to have an exception handler that would transform the 500 message to json).
3. Cannot delete the dynamodb entry for whatever reason (except for non existing which is already catch by `get_item()`) => let crash to return a 500. If the client retry and the file on S3 has already been deleted, this would not cause any issue because of ignoring such failure (see 1.)